scipy/stats/_continuous_distns.py

```diff
@@ -604,6 +605,18 @@ def _cdf(self, x, a, b):
     def _ppf(self, q, a, b):
         return sc.btdtri(a, b, q)
 
+    def _sf(self, x, a, b):
+        return _lazywhere(x < 0.5,
```

Comment:
Currently this uses only `x`, the input, to predict which of the two options will be more accurate. Is the idea to avoid loss of precision in `1 - x` for small `x`, or is this a heuristic that uses `x` to predict whether the `cdf` will be large or small? 

There are cases (e.g. a = 1001, b = 99000 from gh-12794) where `x` may be small, yet already the `cdf` is nearly 1, so this ends up doing `1 - 0.9999999999999999`. If we're shooting for accuracy - not speed - then should we do something like:
```python3
        sf = 1 - sc.btdtr(a, b, x)
        cond = (x > np.finfo(x.dtype).eps) & (sf < 0.5)
        return _lazywhere(cond,
                          (x, a, b),
                          lambda x, a, b: sc.btdtr(b, a, 1 - x),
                          fillvalue=sf)
```

Reply:
The condition `sf < 0.5` is the same as x being greater than the median.  I pushed an update to use the median as the threshold for choosing which expression to evaluate, and that works.  However, I think this PR can be made obsolete if we use Boost to implement the methods, so I'm closing the issue.  We can revisit this PR if it turns out there is a problem using Boost for this, or if the Boost code doesn't make it into 1.7.

Reply:
Yes, I made this comment before that changed was pushed, and that's one of the things I was suggesting. The other suggestion was that if 1 - x == 1, it still may be preferable to evaluate the survival function without performing 1 - x.